logconsumer_test.go

```diff
@@ -508,3 +511,117 @@ func Test_StartLogProductionStillStartsWithTooHighTimeout(t *testing.T) {
 
 	terminateContainerOnEnd(t, ctx, c)
 }
+
+func Test_MultiContainerLogConsumer_CancelledContext(t *testing.T) {
+	// Redirect stderr to a buffer
+	oldStderr := os.Stderr
+	r, w, _ := os.Pipe()
+	os.Stderr = w
+
+	// Context with cancellation functionality for handling interrupt (SIGINT) and
+	// termination (SIGTERM) signals.
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
```

Comment:
@JordanP : 
Thank you for reviewing this PR. The reason for trapping the signal over there was to simulate an interrupt signal `Ctrl+C`, as that was the main motivation behind creating this PR. The objective was to prevent system hangs when multiple containers with log consumers receive an interrupt signal. However, with the latest version, the system hangs if someone presses `Ctrl + C` to stop multiple containers with log consumers.

Reply:
@JordanP : Thanks for your suggestion. Yes, I tried using `context.WithCancel(...)` instead of `signal.NotifyContext(...)` and it seems to have a similar effect in the test case. I have made the changes.